src/Shared/DiagnosticSourceInstrumentation/PropertyFetcher.cs

```diff
@@ -48,28 +49,36 @@ public PropertyFetcher(string propertyName)
     /// </summary>
     /// <param name="obj">Object to be fetched.</param>
     /// <param name="value">Fetched value.</param>
-    /// <param name="skipObjNullCheck">Set this to <see langword= "true"/> if we know <paramref name="obj"/> is not <see langword= "null"/>.</param>
     /// <returns><see langword= "true"/> if the property was fetched.</returns>
 #if NET6_0_OR_GREATER
     [RequiresUnreferencedCode(TrimCompatibilityMessage)]
 #endif
-    public bool TryFetch(object obj, out T? value, bool skipObjNullCheck = false)
+    public bool TryFetch(
+#if NETSTANDARD2_1_0_OR_GREATER || NET6_0_OR_GREATER
+        [NotNullWhen(true)]
+#endif
+        object? obj,
+        out T? value)
     {
-        if (!skipObjNullCheck && obj == null)
+        var innerFetcher = this.innerFetcher;
+        if (innerFetcher == null)
```

Comment:
Use `is null` check here as well?

Reply:
I think in this case the `is` on the `object?` coming in _is_ (pun intended :smile:) important. Because the thing coming in may have operators defined we would want to bypass for best perf. The `==` on `innerFetcher` is more of a style thing because we know it doesn't have operators defined. But I updated it just to be consistent and so it doesn't look like a mistake.